src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -1476,6 +1486,7 @@ std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Pane::_Split(SplitState
     //   Move our control, guid into the first one.
     //   Move the new guid, control into the second.
     _firstChild = std::make_shared<Pane>(_profile.value(), _control);
+    _firstChild->_connectionState = std::exchange(_connectionState, ConnectionState::NotConnected);

```

Comment:
Is it just me or is it kinda hacky that this isn't part of the/a constructor?

Reply:
I rather think the design of Pane leaves a lot to be desired; there are a lot of properties that "move" between parent and leaf panes. The ideal design is one that doesn't require us to move state at all. I'd rather not spend too much effort on an already-bad solution :smile: